src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1039,50 +1039,74 @@ private void RemoveWorseMembers<TMember>(ArrayBuilder<MemberResolutionResult<TMe
 
             var worse = ArrayBuilder<int>.GetInstance(results.Count, unknown);
 
-            for (int c1Idx = 0; c1Idx < results.Count; c1Idx++)
+            int currentBestIndex = -1;
+            for (int index = 0; index < results.Count; index++)
             {
-                var c1result = results[c1Idx];
-
-                // If we already know this is worse than something else, no need to check again.
-                if (!c1result.Result.IsValid || worse[c1Idx] == worseThanSomething)
+                // find the first applicable candidate
+                if (!results[index].Result.IsValid)
                 {
                     continue;
                 }
 
-                bool c1IsBetterThanEverything = true;
-                for (int c2Idx = 0; c2Idx < results.Count; c2Idx++)
+                // assume that the current applicable candidate is the best if we don't have any
+                if (currentBestIndex == -1)
                 {
-                    var c2result = results[c2Idx];
-
-                    if (!c2result.Result.IsValid)
-                    {
-                        continue;
-                    }
-
-                    if (c1result.Member == c2result.Member)
+                    worse[currentBestIndex = index] = betterThanEverything;
```

Comment:
>currentBestIndex = index [](start = 26, length = 24)

Please do not use inline assignment unless there is no way to rewrite the code with assignment statement.